server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketSortPipelineAggregator.java

```diff
@@ -78,7 +79,13 @@ public InternalAggregation reduce(InternalAggregation aggregation, AggregationRe
         for (int i = from; i < limit; ++i) {
             newBuckets.add(ordered.get(i).internalBucket);
         }
-        return originalAgg.create(newBuckets);
+
+        InternalAggregation internalAggregation = originalAgg.create(newBuckets);
+        if (internalAggregation instanceof InternalFilters internalFilters) {
+            return new InternalFilters(internalFilters.getName(), internalFilters.getBuckets(), false, internalFilters.getMetadata());
+        }
+
+        return internalAggregation;
```

Comment:
I'm not super comfortable having one aggregation cause another aggregation to change it's rendering like this. I get that it's a much more sensible way to do it, but it feels weird. I'd prefer to have this be something you configure on the agg up front. Over [here](https://github.com/elastic/elasticsearch/issues/83957#issuecomment-1083317711) we talked about letting folks configure the `keyed` parameter - but I'm not really sure that makes sense....

Both this solution and overriding the `keyed` parameter kind of have the opposite problem - if you don't name the filters and then you sort then you can't figure out which one is which.

It all sort of makes me wonder - why point the bucket_sort pipeline at the filters agg at all? It sort of breaks it.

Finally, I'm a bit worried about changing this json output without some kind of extra parameter because it's *technically* breaking. I mean, the way we're outputting these now is kind of garbage in combination with the `bucket_sort` pipeline agg. But so many folks use Elasticsearch right now, any change in json format is likely to break someone.

So I wonder if the "right" way to fix this is to add a new parameter to the `filters` agg that causes it to output the results in this way. Maybe a `ordered_buckets: true` or  something. I don't know. I'm not good at names. And, yeah, it's lame to need a new parameter. But it's paranoid. And in this case I think that paranoia is good.

Reply:
I agreed, and then I added a keyed_bucket_in_array param as you suggested to avoid judging this in bucket sort.

Now the user can directly modify the return type regardless of whether there is a bucket sort.